test/decl/func/typed_throws.swift

```diff
@@ -103,3 +102,24 @@ func testMapArray(numbers: [Int]) {
     let _: Int = error // expected-error{{cannot convert value of type 'MyError' to specified type 'Int'}}
   }
 }
+
+// Inference of Error conformance from the use of a generic parameter in typed
+// throws.
+func requiresError<E: Error>(_: E.Type) { }
```

Comment:
A -debug-generic-signatures | FileCheck test might be more direct than the requiresError() thing, but it probably doesn't matter